File: implementations/csharp/src/Loom.Core/Models/ModelMetaData.cs
using System.Collections.Generic;

namespace Loom.Core.Models
{
    /// <summary>
    /// Select the best model for a provider
    /// </summary>
    public class ModelMetaData
    {
        public string ModelId { get; set; }
        public string ProviderName { get; set; }
        public int ContextWindow {  get; set; }
        public bool SupportsTools { get; set; }
        public int CostLevel { get; set; } // 1 (cheap) -> 5 (expensive)
        public int LatencyLevel { get; set; } // 1 (fastest) -> 5 (slowest)
        public int QualityLevel { get; set; } // 1 (basic) -> 5 (most capable)
        public List<string> Capabilities { get; set; } // (vision, audio, ...)
    }
}

File: implementations/vb/Loom.Core/Models/ModelMetaData.vb
Namespace Models
    ''' <summary>
    ''' Select the best model for a provider
    ''' </summary>
    Public Class ModelMetaData
        Public Property ModelId As String
        Public Property ProviderName As String
        Public Property ContextWindow As Integer
        Public Property SupportsTools As Boolean
        Public Property CostLevel As Integer ' 1 (cheap) -> 5 (expensive)
        Public Property LatencyLevel As Integer ' 1 (fastest) -> 5 (slowest)
        Public Property QualityLevel As Integer ' 1 (basic) -> 5 (most capable)
        Public Property Capabilities As List(Of String) ' ('vision', 'audio'...)
    End Class
End Namespace